src/statement-creators.ts

```diff
@@ -90,29 +62,24 @@ function createMetadataForTypeInfo(
   if (typeModule) {
     requiredType = `require('${typeModule}').${requiredType}`
   }
-  return f.createObjectLiteralExpression(
-    [
-      f.createPropertyAssignment(
-        'name',
-        f.createStringLiteral(typeInfo.name),
-      ),
-      f.createPropertyAssignment(
-        'type',
-        f.createCallExpression(filterInterfaceFunctionName, undefined, [
-          f.createStringLiteral(requiredType),
-        ])
-      ),
-      f.createPropertyAssignment(
-        'parameters',
-        f.createArrayLiteralExpression(
-          typeInfo.parameters.map((param) =>
-            createMetadataForTypeInfo(f, param, filterInterfaceFunctionName, typesByModule)
-          )
+  const properties: ts.ObjectLiteralElementLike[] = [
+    f.createPropertyAssignment('name', f.createStringLiteral(typeInfo.name)),
+    f.createPropertyAssignment(
+      'type',
+      f.createCallExpression(filterInterfaceFunctionName, undefined, [f.createStringLiteral(requiredType)])
+    ),
+    f.createPropertyAssignment(
+      'parameters',
+      f.createArrayLiteralExpression(
+        typeInfo.parameters.map((param) =>
+          createMetadataForTypeInfo(f, param, filterInterfaceFunctionName, typesByModule)
         )
-      ),
-    ],
-    true
-  )
+      )
+    ),
+  ]
+  if (typeInfo.isNullable) properties.push(f.createPropertyAssignment('isNullable', f.createTrue()))
+  if (typeInfo.isEnum) properties.push(f.createPropertyAssignment('isEnum', f.createTrue()))
```

Comment:
With the above suggestion of having these fields non-optional, this could be simplified (and also the client code)